### ISSUE TITLE
Deploy despite project errors

### ISSUE THREAD

**Issue by Rhobal** [GoogleCloudPlatform/google-cloud-eclipse#2253]
- Cloud Tools for Eclipse Version: 1.2.0.201706082053
- OS: Windows 7 6.1
- Java Version: 1.8.0_131

Cloud Tools won't let me deploy if the project contains errors. In my case, these errors come from a misconfigured plugin; I can delete them from the problem view and start the deploy anyway.

The plugin should only show a warning and allow deployment anyway, it is up to the developer to decide if a deployment makes sense in that situation, not up to the tool

**Comment by chanseokoh** [comment]
Thanks for the suggestion. I think this makes sense. We could prompt if there is an error.

**Comment by elharo** [comment]
Assume a user did attenmpt to deploy a project with a Java syntax error or a malforemd appengine-web.xml. At what point and how would this fail?

**Comment by chanseokoh** [comment]
Currently, a small info dialog will pop up as soon as the user executes the deploy command (e.g., clicks on the deploy context menu item) if there is any error to the project. The user won't see the deploy dialog.

**Comment by elharo** [comment]
The question is not now, but if we implemented this request. It will fail but where and how?

**Comment by chanseokoh** [comment]
I am thinking of changing the current info dialog to a yes/no prompt dialog saying something like "Your project has errors. Do you want to deploy anyway? The behavior is undefined."

**Comment by chanseokoh** [comment]
But if there is a reliable way to determine Java compile errors or malformed XML errors, I think we can always block for that case.

**Comment by elharo** [comment]
I'm not comfortable with undefined behavior here. We might do that, but first we need to know exaclty how this will fail, and make sure we're comfortable with that path.

**Comment by briandealwis** [comment]
I think we should follow Eclipse's approach with launching-with-errors: show there are errors, and let the user decide.  This dialog uses the `MessageDialogWithToggle`, I believe.

![screen shot 2017-08-11 at 10 33 41 am](https://user-images.githubusercontent.com/202851/29217705-94a974a6-7e80-11e7-989b-81962f338fb9.PNG)

**Comment by elharo** [comment]
That depends on what happens next. For example, suppose we deploy a project with errors on top of an existing version? Will it replace and break the running app? We need to answer these ques6tions before we can decide if this feature is feasible.

**Comment by Rhobal** [comment]
I agree with @briandealwis that the user should be shown a warning, but decide in the end. If the app doesn't run, then it sill replace and break the running app. No one should deploy on top of a version with traffic anyway, that's what the version feature of App Engin is for.
In my case, I have additional Eclipse plugins that show errors which do not impact the build. I have to delete them in the Problems view to deploy.
If the appengine-web.xml is malformed, the deployment could probably be refused, but even errors in there could be caused by a misconfigured Eclipse.

**Comment by elharo** [comment]
Right now we depend on errors to block Java 7 deployments and other problems with the runtime. If we were to allow this, we'd need to put a lot more checks in to catch problems like a malformed appengine-web.xml file that really should block deployments.
